### PR TITLE
[Chore] Unit test on SurveyRepo > createResponse

### PR DESCRIPTION
**Pull request by sleepylee**
## What happened 👀

Continue to add more unit test coverage SurveyRepository > createResponse

## Insight 📝

- Refactor code to handle the error case - we must wrap the DioError (lower level) to the app level Error (NetworkError).
- Adding unit test for 2 paths accordingly
 
## Proof Of Work 📹
Code cov ++

### CODE REVIEW

**Review by manh-t — APPROVED** [review]
LGTM 🥦

**Review by luongvo — APPROVED** [review]

**Comment by luongvo** [comment]
@sleepylee can you add me to the contributor list? my review is not recognized 🤓

**Comment by sleepylee** [comment]
something is weird with this PR 🤔 I formatted many times already but it keeps failing wtf 😢

**Comment by luongvo** [comment]
> something is weird with this PR 🤔 I formatted many times already but it keeps failing wtf 😢

@sleepylee you have 1 formatting issue at line 101 💥 
https://github.com/sleepylee/flutter-survey/pull/115#issuecomment-915825025

File: test/repositories/survey_repository_test.dart [hunk 1]
```diff
@@ -96,11 +98,36 @@ main() async {
 
     test('When getSurveyById failed, it throws a NetworkException-NotFound',
         () async {
-      when(mockGraphQLClient.query(any))
-          .thenAnswer((_) async => QueryResult.optimistic(data: null));
+          when(mockGraphQLClient.query(any))
```
Comment (sleepylee) [inline test/repositories/survey_repository_test.dart:101]:
```suggestion
      when(mockGraphQLClient.query(any))
```

**Comment by sleepylee** [comment]
@luongvo of course I read the error message 😓 but sth is wrong...I did format + commit but it keeps messing up..

**Review by luongvo — APPROVED** [review]

**Pull request merged**